### PR TITLE
Add tailing metrics

### LINKED ISSUE #6027: Additional metrics for monitoring tailer component
**Is your feature request related to a problem? Please describe.**

At the moment there is no good way to monitor log tailing connections.

**Describe the solution you'd like**

Add new metrics for:
* active tail connection
* amount of active streams tailed
* amount of data sent to clients

**Describe alternatives you've considered**

Some of the metrics could be derived from gRPC connection logs.

### PR DESCRIPTION
**Pull request by MasslessParticle**
Add tail metrics.

Fixes #6027

### CODE REVIEW

File: pkg/querier/tail.go [hunk 1]
```diff
@@ -106,6 +114,7 @@ func (t *Tailer) loop() {
 				continue
 			}
 
+			entriesSize += len(t.currEntry.Line)
```
Comment (MasslessParticle) [inline pkg/querier/tail.go:117]: This isn't necessarily the number of bytes, but it is consistent with how we calculate similar metrics in other components.

**Review by trevorwhitney — APPROVED** [review]
LGTM!

File: pkg/querier/tail.go (hunk 1 above)
Comment (MasslessParticle) [reply pkg/querier/tail.go:117]: You're right: Golang `len` gives back bytes rather characters. I can't believe I didn't know this.

I wonder how many bugs I've caused with it?

Comment (trevorwhitney) [reply pkg/querier/tail.go:117]: haha, go was clearly designed for dummies like me who only use English characters and still think ASCII is the only encoding out there.

File: pkg/querier/tail.go [hunk 2]
```diff
@@ -264,25 +279,48 @@ func (t *Tailer) getCloseErrorChan() <-chan error {
 	return t.closeErrChan
 }
 
+func (t *Tailer) recordStream(id uint64) {
+	t.seenStreamsMtx.Lock()
+	defer t.seenStreamsMtx.Unlock()
+
+	if _, ok := t.seenStreams[id]; ok {
```
Comment (cyriltovena) [inline pkg/querier/tail.go:286]: If you're only writing once there's a lot of chances that `sync.Map` will perform better. Not sure this is important in this case but at least I shared the options for later.

**Review by cyriltovena — APPROVED** [review]
LGTM

**Pull request merged**